### ISSUE TITLE
Wiring Questions

### ISSUE THREAD

**Issue by stevenjev** [nervalo88/ESP8266_INTEX_SPA#1]
I've been wanting to do something like this on my Intex spa for a while, and just found your project. This is a fantastic reference and will definitely help me with my own project! Just a couple questions. Soldering to R3,R4 and R5 - are these to the LEDs on the controller (as in you are monitoring the states by determining if the LED is on or off)? And if so, are you just using adding a single wire to each indicated location, and then the common ground is the one on the main board, near the +5V header? If you had more photos or a wiring sketch that would be great, but no worries if not.

Additionally, if I am correct that you are just determining status through the LED, are you able to differentiate if the heater is on and heating (red LED) versus if the heater is on and not heating (green LED).

Thanks again for posting all of these details.

**Comment by stevenjev** [comment]
Actually, I found the additional photos, my apologies! I see that you are wiring directly to the mainboard. Does this mean you are basically pulling off those resistors to see if the controller has engaged the relays?

**Comment by nervalo88** [comment]
Hello,
Thanks for your encouraging words.
indeed, the controller leds are multiplexed into a serial bus (7HC595) which make them uneasy to get as the same bus pins is sometimes also polled to get the button states.  

My strategy after compiling all the existing work on the web was to :
- get the state of the the power board existing relays by picking relays control signals from the spa µcontroller. (level converter impedance does not change the operation)- in other word the ESP is in parrallel, with its input pins joined with spa output pins
- control the button presses directly on the controller via additionnal relays (not to perturbate the multiplexing/ polling)

I aimed to be the less invasive as possible, to keep the original safeties.

**Comment by stevenjev** [comment]
Apologize for the delay responding. Thank you for the detailed response, it has helped me to finish my project. However, I am running into an issue and am wondering if you have encountered these two issues:

When the outside air heats up (warm summer day), the d1 mini overheats and shuts off. This does not happen when the air temperature is cold. Have you had this problem?

And two, when it does overheat, it also prevents the hot tub heater from properly heating. The display states that the heater is on, however the heater relay shuts off when the d1 mini overheats. I believe this may be related to using the main board pin headers to power the d1 mini as those pins do control functions of the hot tub. Curious if you have had this problem?

Thanks again!

**Comment by Listwa** [comment]
Hi,
Not seeing that problem. I went another way.  I used a ESP8266 4 channel
wifi relay board attached to the ON, Heater and Temp up and down for
control.  I powered it by picking up 5v right in the control head. To
monitor Temp,  I am using  a La Crosse Alerts Mobile Wireless 926-251031-HT
Monitor System with Hot Tub Accessory Set
<https://www.amazon.com/gp/product/B0081URKJU/ref=ppx_yo_dt_b_asin_title_o05_s00?ie=UTF8&psc=1>
( $32).   I am also able to monitor power usage with  a smart plug with
power usage monitoring.  So yes, I need to run 3 apps, that currently
dont talk to each other, but the systems work and I can turn things on and
off.   What is interesting is that currently the Spa is using about $1 per
day in electrical usage if I allow it to stay at 104 degrees all day.

**Comment by stevenjev** [comment]
Are you storing the esp8266 within the hot tub controller housing? No overheat issues?

**Comment by Listwa** [comment]
I am storing it in the top 1/2.  There is the head with  the Display, then
the 2 foot shaft,  which attaches to the main unit.   I store it at the
bottom of the 2 foot shaft, about 1 inch from the bottom. I wedged it up
there, then hot glued it.

**Comment by stevenjev** [comment]
Ah that may be my problem, that sounds like a better spot for it. I’ve attached mine to the main control board cover and it just gets too warm in there. Thanks!

**Comment by Listwa** [comment]
The key was tapping 5v from the display board.

**Comment by stevenjev** [comment]
Ah so you don’t use the header pins from the main board like described in this project?

**Comment by Listwa** [comment]
nope, but traced where it powers the control board and confirmed with a
voltmeter it was 5V.  I am using EasyTCP to send the commands.  Only issues
is the relay/ apps dont " inch"  and i need to do a command on , then a
command off to toggle the control. There are no smarts ,  I need to
manually check Temp and raise or lower it as needed or turn on the heater.
 I can tell when the units on as it measures 3 W, when the pump is on its
about 46W and when the heaters on its about 1400W.
